parcels/particleset.py

```diff
@@ -496,16 +496,20 @@ def density(self, field=None, particle_val=None, relative=False, area_scale=Fals
 
         :param field: Optional :mod:`parcels.field.Field` object to calculate the histogram
                       on. Default is `fieldset.U`
-        :param particle_val: Optional numpy-array of values to weigh each particle with.
-                             Default is 1 for each particle
+        :param particle_val: Optional numpy-array of values to weigh each particle with,
+                             or string name of particle attribute to use weigh particles with.
+                             Default is None, resulting in a value of 1 for each particle
         :param relative: Boolean to control whether the density is scaled by the total
                          weight of all particles. Default is False
         :param area_scale: Boolean to control whether the density is scaled by the area
                            (in m^2) of each grid cell. Default is False
         """
 
         field = field if field else self.fieldset.U
-        particle_val = particle_val if particle_val else np.ones(len(self.particles))
+        if isinstance(particle_val, basestring):
```

Comment:
We do want to be python3 compatible (see #281), so better to use `if isinstance(particle_val, (str, unicode))` here?